packages/gantt/frontend/sirius-components-gantt/src/helper/helper.tsx

```diff
@@ -180,3 +189,87 @@ export const getDisplayedColumns = (gqlColumns: GQLColumn[]): Column[] => {
 export const getSelectedColumns = (gqlColumns: GQLColumn[]) => {
   return gqlColumns.filter((col) => col.displayed === true).map((col) => TaskListColumnEnum[col.id]);
 };
+
+const getDayOfTheYear = (date: Date) => {
+  const start = new Date(date.getFullYear(), 0, 0);
+  const diff = date.getTime() - start.getTime() + (start.getTimezoneOffset() - date.getTimezoneOffset()) * 60 * 1000;
+  const oneDay = 1000 * 60 * 60 * 24;
+  const day = Math.floor(diff / oneDay);
+  return day;
+};
+
+export const roundDate = (
+  date: Date,
+  dateExtremity: DateExtremity,
+  action: BarMoveAction,
+  dateMoveStep: String
+): Date => {
+  const regex = /^(\d+)([DHm])$/;
```

Comment:
I don't see how asking for any string on the backend and parsing that with a regex on the frontend provides an usable API. Don't use regex!

Reply:
I needed a simple solution for this time frame.
The string is composed of an integer and a letter (corresponding to the time unit from ISO 8601).
It will be explained in the specifier documentation.
If the string does not follow the expected syntax, it will fall back to 1 day date rounding.
But I agree with you, we should ensure a good syntax on backend (rule at saving?, validation rule?) I did not think about this part.
On the other hand, I preferred a single field to avoid shattering information in the GanttDescription and the ideal solution of having a field (value + dimension) was an enhancement in and of itself.

Reply:
I changed the graphQL API two have a type `GanttDateRounding` composed of an integer value and an enum